src/dali/abstract_pair_converter_plugin.py

```diff
@@ -179,35 +227,28 @@ def _build_fiat_list(self) -> None:
             LOGGER.info("Fetching of fiat symbols failed. The server might be down. Please try again later.")
             raise RP2RuntimeError("JSON decode error") from exc
 
-    def _add_fiat_edges_to_graph(self, graph: Dict[str, Dict[str, None]], markets: Dict[str, List[str]]) -> None:
+    def _add_fiat_edges_to_graph(self, graph: MappedGraph[str], markets: Dict[str, List[str]]) -> None:
         if not self.__fiat_list:
             self._build_fiat_list()
 
         for fiat in self.__fiat_list:
             to_fiat_list: Dict[str, None] = dict.fromkeys(self.__fiat_list.copy())
             del to_fiat_list[fiat]
-            if graph.get(fiat):
+            # We don't want to add a fiat vertex here because that would allow a double hop on fiat (eg. USD -> KRW -> JPY)
+            if graph.get_vertex(fiat):
                 for to_be_added_fiat in to_fiat_list:
-                    # add a pair if it doesn't exist
-                    if to_be_added_fiat not in graph[fiat]:
-                        graph[fiat][to_be_added_fiat] = None
-            else:
-                graph[fiat] = to_fiat_list
+                    graph.add_neighbor(
+                        fiat,
+                        to_be_added_fiat,
```

Comment:
fiat to fiat is fine and necessary.
fiat to fiat to fiat is what I was trying to avoid, and that is what I was calling a double hop.

Basically, I don't want nodes from the fiat 'exchange' linking to each other. I only want them to link to the fiat on the crypto exchanges.